api/src/main/java/com/eleks/academy/pharmagator/services/MedicineControllerService.java

```
@@ -0,0 +1,58 @@
+package com.eleks.academy.pharmagator.services;
+
+import com.eleks.academy.pharmagator.entities.Medicine;
+import com.eleks.academy.pharmagator.repositories.MedicineRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class MedicineControllerService implements ControllerService<Long, Medicine, Medicine> {
+    private final MedicineRepository medicineRepository;
+
+    @Override
+    public ResponseEntity<List> getAll() {
+        return ResponseEntity.ok(medicineRepository.findAll());
+    }
+
+    @Override
+    public ResponseEntity<Medicine> get(Long id) {
+        Optional<Medicine> medicine = medicineRepository.findById(id);
+
+        if (!medicine.isPresent()) {
+            return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
```

It would be better to throw an exception here